src/xcode/ENA/ENA/Source/Services/CCLService/CCLService.swift

```diff
@@ -18,29 +21,104 @@ enum DCCWalletInfoAccessError: Error {
 
 protocol CCLServable {
 	
-	func updateConfiguration(completion: (Swift.Result<Void, CLLServiceError>) -> Void)
+	func updateConfiguration(completion: @escaping (_ didChange: Bool) -> Void)
 	
 	func dccWalletInfo(for certificates: [DCCWalletCertificate]) -> Swift.Result<DCCWalletInfo, DCCWalletInfoAccessError>
 	
 	func evaluateFunctionWithDefaultValues<T: Decodable>(name: String, parameters: [String: AnyDecodable]) throws -> T
-	
-	var configurationDidChange: PassthroughSubject<Bool, Never> { get }
 
 }
 
-class CLLService: CCLServable {
+class CCLService: CCLServable {
 	
 	// MARK: - Init
 	
-	init() { }
+	init(
+		_ restServiceProvider: RestServiceProviding
+	) {
+		self.restServiceProvider = restServiceProvider
+
+		// boosterNotificationRules
+		switch restServiceProvider.cached(boosterNotificationRulesResource) {
+		case let .success(rules):
+			self.boosterNotificationRules = rules.rules
+		case let .failure(error):
+			Log.error("Fialed to load boosterNotification rules from cache - init them empty", error: error)
+			self.boosterNotificationRules = []
+		}
+
+		// cclConfigurations
+		switch restServiceProvider.cached(cclConfigurationResource) {
+		case let .success(configurations):
+			self.cclConfigurations = configurations.cclConfigurations
+		case let .failure(error):
+			Log.error("Failed to read ccl configurations from cache - init empty", error: error)
+			self.cclConfigurations = []
+		}
+	}
 	
 	// MARK: - Protocol CCLServable
 
-	func updateConfiguration(completion: (Swift.Result<Void, CLLServiceError>) -> Void) {
-		completion(.success(()))
+	func updateConfiguration(
+		completion: @escaping (_ didChange: Bool) -> Void
+	) {
+		// trigger both downloads, if one was updated notify caller in result
+
+		let dispatchGroup = DispatchGroup()
+
+		var configurationDidUpdate: Bool = false
+		var boosterRulesDidUpdate: Bool = false
+
+		// lookup configuration updates
+		dispatchGroup.enter()
+		getConfigurations { [weak self] result in
+			defer {
+				dispatchGroup.leave()
+			}
+
+			switch result {
+			case let .success(configurations):
+				// we got a new configuration - let update json functions
+				configurations.forEach { configuration in
+					self?.updateJsonFunctions(configuration)
+				}
+				self?.cclConfigurations = configurations
+				configurationDidUpdate = true
+			case let .failure(error):
+				if case let .cached(configurations) = error {
+					self?.cclConfigurations = configurations
+					configurationDidUpdate = false
+				}
+			}
+		}
+
+		// lookup booster notification rules updates
+		dispatchGroup.enter()
+		getBoosterNotificationRules { [weak self] result in
+			defer {
+				dispatchGroup.leave()
+			}
+
+			switch result {
+			case let .success(rules):
+				self?.boosterNotificationRules = rules
+				boosterRulesDidUpdate = true
+			case let .failure(error):
+				if case let .cached(rules) = error {
+					self?.boosterNotificationRules = rules
+					boosterRulesDidUpdate = false
+				}
```

Comment:
Now that we initially read the cache I think we could also not do anything at all here